BuildScript/BuildTasks/CreateTestConfigurations.cs

```diff
@@ -0,0 +1,180 @@
+          supportedConfigurationIDs: s_supportedConfigurationIDs,
+          supportedExecutionRuntimes: GetMetadataValueDictionary (
+              SupportedExecutionRuntimes,
+              MetadataValueConstants.EnforcedLocalMachine,
+              MetadataValueConstants.LocalMachine),
+          supportedTargetRuntimes: GetMetadataValueCollection (SupportedTargetRuntimes));
+
+      var testConfigurationsFilter = new TestConfigurationsFilter (
+          supportedBrowsers: GetMetadataValueCollection (SupportedBrowsers, MetadataValueConstants.NoBrowser),
+          enabledBrowsers: GetMetadataValueCollection (EnabledBrowsers),
+          supportedDatabaseSystems: GetMetadataValueCollection (SupportedDatabaseSystems, MetadataValueConstants.NoDB),
+          enabledDatabaseSystems: GetMetadataValueCollection (EnabledDatabaseSystems),
+          supportedPlatforms: GetMetadataValueCollection (SupportedPlatforms),
+          enabledPlatforms: GetMetadataValueCollection (EnabledPlatforms),
+          supportedExecutionRuntimes: GetMetadataValueDictionary (
+              SupportedExecutionRuntimes,
+              MetadataValueConstants.EnforcedLocalMachine,
+              MetadataValueConstants.LocalMachine),
+          enabledExecutionRuntimes: GetMetadataValueCollection (EnabledExecutionRuntimes),
+          supportedTargetRuntimes: GetMetadataValueCollection (SupportedTargetRuntimes),
+          enabledTargetRuntimes: GetMetadataValueCollection (EnabledTargetRuntimes),
+          supportedConfigurationIDs: s_supportedConfigurationIDs,
+          enabledConfigurationIDs: GetMetadataValueCollection (EnabledConfigurationIDs));
+
+      var output = new List<ITaskItem>();
+      foreach (var inputItem in Input)
+      {
+        var testAssemblyFullPath = inputItem.ItemSpec;
+        var testConfigurationsMetadataValue = inputItem.GetMetadata ("TestConfiguration");
+        var testSetupBuildFile = inputItem.GetMetadata ("TestSetupBuildFile");
+
+        if (string.IsNullOrEmpty (testConfigurationsMetadataValue))
+        {
+          Log.LogError ($"Could not find TestConfigurations for {testAssemblyFullPath}.");
+          return false;
+        }
+
+        var rawTestConfigurations = testConfigurationsMetadataValue
+            .Split (new[] { ";" }, StringSplitOptions.RemoveEmptyEntries)
+            .Select (x => x.Trim())
+            .ToArray();
+
+        try
+        {
+          var unfilteredTestConfigurations = testConfigurationsFactory.CreateTestConfigurations (
+              testAssemblyFullPath,
+              rawTestConfigurations,
+              testSetupBuildFile);
+          var filteredTestConfigurations = testConfigurationsFilter.GetFilteredTestConfigurations (unfilteredTestConfigurations);
+          output.AddRange (filteredTestConfigurations.Select (CreateTestConfigurationTaskItem));
+        }
+        catch (Exception ex) when (ex is InvalidOperationException || ex is ArgumentException)
+        {
```

Comment:
Am not sure I like the exception test this way. It feels...wrong. Usually, there are two separate exception blocks. Doing it this way (i.e. catching the Exception and then filtering on the type) just doesn't feel idiomatic. Maybe it's different when the catch-block contains more code...

Reply:
I've split it into separate catch blocks but I'm not sure if I prefer the slight duplication over the pattern match. Please tell me what you think once I request a new review